### ISSUE TITLE
Can't build PowerPC SPE toolchain for glibc 2.8

### ISSUE THREAD

**Issue by electricworry** [crosstool-ng/crosstool-ng#644]
I've got an issue trying to build a legacy toolchain. Shoot me down if I should not reasonably expect the following configuration to be supportable today, but I would quite like the following to work if possible as I'm dealing with an old embedded system with these properties:

Base configuration: powerpc-e500v2-linux-gnuspe
Modifications (using ct-ng menuconfig):
* C-library > glibc version > 2.8 (version on target)
* C-library > Minimum supported kernel version > 2.6.21 (again, this is the version on target)

The following changes also need to be made to avoid errors about as, ld, gcc, make all being "too old". (Rather they are "too new", as the C-library prerequisites didn't anticipate significantly newer versions.)
* C compiler > gcc version > 4.7.4
* Binary utilities > binutils version > 2.19.1a
* Companion tools > Build some companion tools > make

I get the error as follows:
```
[ALL  ]    /usr/bin/make  subdir=manual -C manual ..=../ subdir_lib
[ALL  ]    make[3]: Entering directory '/home/bob/ppc/.build/src/glibc-2.8/manual'
[ALL  ]    Makefile:235: *** mixed implicit and normal rules: deprecated syntax
[ALL  ]    make[3]: *** No rule to make target '/home/bob/ppc/.build/powerpc-e500v2-linux-gnuspe/build/build-libc-final/manual/stamp.o', needed by 'lib'.  Stop.
[ALL  ]    make[3]: Leaving directory '/home/bob/ppc/.build/src/glibc-2.8/manual'
[ALL  ]    Makefile:206: recipe for target 'manual/subdir_lib' failed
[ERROR]    make[2]: *** [manual/subdir_lib] Error 2
[ALL  ]    make[2]: Leaving directory '/home/bob/ppc/.build/src/glibc-2.8'
[ALL  ]    Makefile:12: recipe for target 'all' failed
[ERROR]    make[1]: *** [all] Error 2
[ALL  ]    make[1]: Leaving directory '/home/bob/ppc/.build/powerpc-e500v2-linux-gnuspe/build/build-libc-final'
[ERROR]  
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Installing C library'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@257]
[ERROR]  >>        called from: do_libc_backend_once[scripts/build/libc/glibc.sh@470]
[ERROR]  >>        called from: do_libc_backend[scripts/build/libc/glibc.sh@153]
[ERROR]  >>        called from: do_libc[scripts/build/libc/glibc.sh@75]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@646]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      '/home/bob/crosstool-ng/share/doc/crosstool-ng/crosstool-ng-1.22.0/B - Known issues.txt'
[ERROR]  
[ERROR]  (elapsed: 16:14.46)
```

I provided the full build log at https://electricworry.net/public/persistent/ctng-ppcspe-issue-build.log. I am using Ubuntu 16.04 amd64 and crosstool-ng release 1.22.0. I have tried this on a pristine installation.

Think this can be fixed? Pointed to resolve welcome. Spent a long time trying to resolve this, and it would help immensely to resolve.

Thanks.

**Comment by stilor** [comment]
I tried building with your configuration and it first failed while building PPL - I assume you have already figured out that the host gcc 6.x is too new for it. Adding gcc-4.9/g++-4.9 to path so that they precede gcc 6.x helps with that.

The glibc failure is a result of a bug in 1.22 - even though it built make 3.81, it still used host's make (i.e. make 4.x). Edit scripts/build/libc/glibc.sh and replace `${make}` with `make`.

You also need to set the "Enable ports addon" option in "C library" menu.

With those changes, your config built ok.

**Comment by electricworry** [comment]
Hello,

Thanks for the quick response. Great, I changed ${make} to make on line 470 of glibc.sh and the "Building C library" step completed.

Just for reference for anyone else building the same toolchain, I didn't have an issue with PPL, as my version of gcc/g++ was 5.4.0. However, when I reached the "Building final gcc" stage, I ran into problems. Adding gcc-4.9/g++-4.9 as stilor recommended (and adding gcc/g++ symlinks higher in path) resolved this issue. 

The toolchain builds completely now! Thanks. This is an excellent project.
